src/protocol/prf_sharding/mod.rs

```diff
@@ -105,22 +141,38 @@ impl InputsRequiredFromPrevRow {
                 &self.attributed_breakdown_key_bits,
                 &bd_key,
             ),
+            timestamp_of_most_recent_source_event(
+                ctx.narrow(&Step::SourceEventTimestamp),
+                record_id,
+                attribution_window_seconds,
+                &input_row.is_trigger_bit,
+                &self.source_event_timestamp,
+                &timestamp,
+            ),
         )
         .await?;
 
-        let did_trigger_get_attributed = input_row
-            .is_trigger_bit
-            .multiply(
+        let (did_trigger_get_attributed, is_trigger_within_window) = try_join(
+            input_row.is_trigger_bit.multiply(
                 &ever_encountered_a_source_event,
                 ctx.narrow(&Step::DidTriggerGetAttributed),
                 record_id,
-            )
-            .await?;
+            ),
+            is_trigger_event_within_attribution_window(
+                ctx.narrow(&Step::CheckAttributionWindow),
+                record_id,
+                attribution_window_seconds,
+                &timestamp,
+                &source_event_timestamp,
+            ),
+        )
+        .await?;
 
         let attributed_trigger_value = zero_out_trigger_value_unless_attributed(
             ctx.narrow(&Step::AttributedTriggerValue),
             record_id,
             &did_trigger_get_attributed,
+            &is_trigger_within_window,
```

Comment:
An alternative approach, which avoids the unnecessary multiplication in the case of no windowing, is to just lift that multiplication out of the `zero_out_trigger_value_unless_attributed` function and do it here, like this:

```
let zero_out_flag = if Some(window) = attribution_window_seconds {
    let c = ctx.narrow(&Step::AttributedEventCheckFlag);
    &did_trigger_get_attributed
        .multiply(is_trigger_within_window, c, record_id)
        .await?
} else {
    did_trigger_get_attributed
}

let attributed_trigger_value = zero_out_trigger_value_unless_attributed(
    ctx.narrow(&Step::AttributedTriggerValue),
    record_id,
    &zero_out_flag,
    &tv,
)
.await?;
```

Reply:
I added the `attribution_window_seconds` to `zero_out_...()` and do it inside the function instead. I wanted to move both `did_trigger_get_attributed` and `is_trigger_within_window` flags inside the function too, but we need `did_trigger_get_attributed` to mutate `self` at the end of `compute_row_with_previous`.

@benjaminsavage, do we plan to use `did_trigger_get_attributed` in the output? Currently we set it in `CappedAttributionOutputs` but never reference it.